src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -59,9 +59,11 @@ internal FileChangeVersionOne CreateFileChange(FileChange v2FileChange)
 
             if (v2FileChange != null)
             {
+                string encodingName = GetFileEncodingName(v2FileChange.FileLocation?.Uri);
```

Comment:
I understand the trade-off you're making. You call `Encoding.GetEncoding(encodingName)` on every replacement, rather than doing it once, so you have `encodingName` available to throw in the exception.

As an alternative, you could call `GetEncoding` once, and pass _both_ `encoding` and `encodingName` into `CreateReplacement`.  Your call.

Reply:
Yeah good point, I don't like the repetition. What do you think about going back to the CreateReplacement(r, encoding) model, catching our UnknownEncodingException in CreateFileChange, setting the encoding name, and re-throwing?

Reply:
I think that's fine as long as there's a comment explaining why you're doing the catch and rethrow.